Sources/TripKit/managers/TKGeoMonitorManager.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  TKGeoMonitorManager.swift
+//  TripKitUI-iOS
+//
+//  Created by Jules Ian Gilos on 1/12/23.
+//  Copyright © 2023 SkedGo Pty Ltd. All rights reserved.
+//
+
+import Foundation
+import GeoMonitor
+import CoreLocation
+
+import UserNotifications
+
+/// The manager for geofence-based alerts for a trip, e.g., "get off at the next stop"
+///
+/// Requirements:
+/// - Project > Your Target > Capabilities > Background Modes: Enable "Location Updates"
+/// - Project > Your Target > Info: Include both `NSLocationAlwaysAndWhenInUseUsageDescription` and `NSLocationWhenInUseUsageDescription`
+public class TKGeoMonitorManager: NSObject {
```

Comment:
Another requirement is that `TKNotificationManager` has a publisher. Should log a warning if `setAlertsEnabled(_:trip:)` is called but there's no publisher listening for these.

Alternatively, we could disable this feature if no publisher is set. And use that as the mechanism to opt into this feature.

Reply:
shouldn't opt-in be in `TGTripOverviewCardCustomizer`? though I'm still looking into it and it requires TGCards. 

Adding `TKNotificationManager` as a Notification DataSource seems broad and confusin but it works. I'm still unsure with that.

Reply:
added `TKNotificationManager` with context so it's easier to identify, also displays the notification toggle when notification data source is subscribed

Reply:
Nice, that's a good way.